### ISSUE TITLE
Reset when working on Kata does not reset tests, only code.

### ISSUE THREAD

**Issue by ChristianECooper** [Codewars/codewars.com#106]
It's all in the title!

If you work on a kata, modify the code and modify the example tests, then hit reset and confirm, only the code is reset, the tests are not reset to their base state.

**Comment by paul-calvelage** [comment]
This may not be labeled as an actual "bug" from a development standpoint, but from a UI standpoint it sure feels like one.  There is a button that says "Reset" but does not fully reset.

I tend to think that anytime the software does not do what I expect, it is a bug.  On the other hand, different people have different expectations when it comes to UI.  Does that make every difference of opinion a bug?

Nevertheless, when I clicked reset the first few times, I expected it to also reset the test cases.  From experience I have learned it is unable do that.  Call it a missing feature or a bug as you will.

Sometimes the test cases are the only way to know what the author wants the result to be.  If you lose the tests, you are out of luck.  Then you must reverse engineer the test suite for clues.  Judicious logging is helpful with this.  Or you can hack something together that passes but may not be in the spirit of the kata, in order to start over with the original tests.  I don't cheat the kata, but that is definitely another option to start over with the tests.

Bottom line: don't mess up the tests -- you can't get them back!

**Comment by jon-a-nygaard** [comment]
+1

**Comment by geekytime** [comment]
I cleared out the sample tests, but then I didn't remember which of mocha's silly sets of global asserts they were using.
